src/tests/tests/background.js

```diff
@@ -184,87 +144,102 @@
       this.tabId = -1;
       badger.tabData[this.tabId] = {
         frames: {},
-        origins: {},
-        blockedCount: 0
+        origins: {}
       };
     },
   },
   function() {
+    let SITE_URL = "http://example.com/";
+
     QUnit.module("logThirdPartyOriginOnTab", {
       beforeEach: function () {
-        this.before = beforeMock(['badger.updateBadge']);
-        badger.updateBadge = noop;
+        sinon.stub(badger, "updateBadge");
       },
       afterEach: function () {
-        unmock(this.before);
+        badger.updateBadge.restore();
       },
     });
+
     QUnit.test("increment blocked count", function(assert) {
       let tabId = this.tabId;
-      assert.equal(badger.tabData[tabId].blockedCount, 0);
+      assert.equal(badger.getBlockedOriginCount(tabId), 0);
 
+      badger.storage.setupHeuristicAction('stuff', constants.BLOCK);
       badger.logThirdPartyOriginOnTab(tabId, 'stuff', constants.BLOCK);
-      assert.equal(badger.tabData[tabId].blockedCount, 1);
+      assert.equal(badger.getBlockedOriginCount(tabId), 1);
 
       badger.logThirdPartyOriginOnTab(tabId, 'stuff', constants.BLOCK);
-      assert.equal(badger.tabData[tabId].blockedCount, 1);
+      assert.equal(badger.getBlockedOriginCount(tabId), 1);
 
+      badger.storage.setupHeuristicAction('eff.org', constants.COOKIEBLOCK);
       badger.logThirdPartyOriginOnTab(tabId, 'eff.org', constants.COOKIEBLOCK);
-      assert.equal(badger.tabData[tabId].blockedCount, 2);
+      assert.equal(badger.getBlockedOriginCount(tabId), 2);
     });
+
     QUnit.module('updateBadge', {
       beforeEach: function() {
-        this.before = beforeMock([
-          'chrome.tabs.get',
-          'chrome.browserAction.setBadgeText',
-          'chrome.browserAction.setBadgeBackgroundColor'
-        ]);
+        // stub chrome.tabs.get manually as we have some sort of issue stubbing with Sinon in Firefox
+        this.chromeTabsGet = chrome.tabs.get;
```

Comment:
I'm fine with removing these here, but do you mind if I include them in future tests? I generally prefer helper methods if they are short and simple, over 3rd party dependencies.  Especially with Sinon,  I can never tell if it is buggy, or if I just don't understand how it works, or if my code is just broken.

Reply:
My issue with your helpers is that I don't know what they do. Documenting and adding tests would help, but now we're reimplementing parts of Sinon.

It is unfortunate that Sinon and Firefox have some sort of (maybe) WebExtensions-related bug, but I still think it's worth trying to use a more standard solution whenever we can. Sinon's docs are pretty confusing, but there are articles/tutorials/Q&As across the Web. Other open source projects use Sinon so we can look for real-world examples that way. Why get into the business of testing helper libraries?

Also, I tried to keep my workarounds as simple as possible. Do we actually need a helper to do something, or can we wait to add the helper later when we do need it? I would rather keep it simple and wait.

Reply:
> Why get into the business of test helper libraries?

This is not my intention. The purpose of most of the code I wrote was to automate mocking and unmocking a list of functions. Not to entirely replace sinon. You could modify it to use sinon if you'd like. Below it looks like you manually reproduce what the test helper was doing. Where you mock, and then manually unmock, `chrome.tabs.get` and `chrome.browserAction.setBadgeBackgroundColor`, you also have to do the same when you mock functions with sinon. I dislike this pattern because we have to manually track mocking and unmocking (whether it is with sinon or a manual mock). The code I wrote makes this pattern nicer.

>  Do we actually need a helper to do something, or can we wait to add the helper later when we do need it?

No we definitely don't need the code I wrote. I don't think we will ever need it because it can be done manually. I just thought it made writing tests easier. It is a matter of preference to me.

Reply:
I think Sinon supports [automatic unmocking](http://sinonjs.org/releases/v2.0.0/sandbox/). It doesn't seem worth getting into yet since keeping track of three stubs is easier and simpler than getting up to speed with this functionality, but it's there.

Reply:
Would you like me to add documentation for the code I added? I personally think it is easier and simpler than tracking stubs.